tracer/src/Datadog.Trace/Activity/Handlers/DefaultActivityHandler.cs

```diff
@@ -83,10 +83,14 @@ public void ActivityStarted<T>(string sourceName, T activity)
 
                 // We convert the activity traceId and spanId to use it in the
                 // Datadog span creation.
-                traceId ??= Convert.ToUInt64(w3cActivity.TraceId.Substring(16), 16);
-                spanId = Convert.ToUInt64(w3cActivity.SpanId, 16);
-                rawTraceId = w3cActivity.TraceId;
-                rawSpanId = w3cActivity.SpanId;
+                if (w3cActivity.TraceId is { } w3cTraceId)
+                {
+                    // If the Activity has an ActivityIdFormat.Hierarchical instead of W3C it's TraceId & SpanId will be null
+                    traceId ??= Convert.ToUInt64(w3cTraceId.Substring(16), 16);
+                    spanId = Convert.ToUInt64(w3cActivity.SpanId, 16);
```

Comment:
Should we check both properties to be safe?

```suggestion
                if (w3cActivity.TraceId is { } w3cTraceId && w3cActivity.SpanId is { } w3cSpanId)
                {
                    // If the Activity has an ActivityIdFormat.Hierarchical instead of W3C it's TraceId & SpanId will be null
                    traceId ??= Convert.ToUInt64(w3cTraceId.Substring(16), 16);
                    spanId = Convert.ToUInt64(w3cSpanId, 16);
```

Reply:
Yeah makes sense to me, thanks!